jaxrs-clients/src/test/java/com/palantir/remoting3/jaxrs/JaxRsClientFailoverTest.java

```diff
@@ -24,54 +24,51 @@
 import com.google.common.collect.Lists;
 import com.palantir.remoting3.clients.ClientConfiguration;
 import feign.RetryableException;
+import java.io.IOException;
+import java.time.Duration;
 import java.util.List;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import okhttp3.mockwebserver.MockResponse;
 import okhttp3.mockwebserver.MockWebServer;
-import org.junit.Before;
-import org.junit.Rule;
 import org.junit.Test;
+import org.junit.experimental.theories.DataPoint;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
 
-
+@RunWith(Theories.class)
```

Comment:
Woah I'd never seen this before! Does it have any particular advantages over `@Parameterized`?

Reply:
I haven't used `@RunWith(Parameterized.class)` but it seems like `Theories.class` allows for a mix of DataPoint/Theory-style tests and normal `@Test`s (which is beneficial here).